patches/tModLoader/Terraria/HitTile.cs.patch

```diff
@@ -17,3 +17,18 @@
  				flag3 = false;
  			}
  			else if (Main.tileSolid[tile.type] && !Main.tileSolidTop[tile.type]) {
+@@ -346,6 +_,14 @@
+ 
+ 			int animationTimeElapsed = data[j].animationTimeElapsed;
+ 			if (!((float)animationTimeElapsed >= 10f)) {
++				/* Check if the current tile is a modded tile. */
++				if (TileLoader.GetTile(tile.type) is { } modTile) {
++					/* If it is a modtile, custom drawing is handled and we should skip the rest of this loop for this tile. */
++					if (modTile.DrawCracks(spriteBatch, x, y, damage, data[j].crackStyle, true, animationTimeElapsed)) {
```

Comment:
you should probably pass in the default animation `color` and `scale` parameters, as if people just want to change the texture they're going to struggle